addons/hr_expense/models/hr_expense.py

```diff
@@ -418,13 +418,15 @@ def _get_default_expense_sheet_values(self):
             raise UserError(_("You cannot report expenses for different companies in the same report."))
 
         todo = self.filtered(lambda x: x.payment_mode=='own_account') or self.filtered(lambda x: x.payment_mode=='company_account')
+        expense_name = False
         if len(todo) == 1:
             expense_name = todo.name
         else:
             dates = todo.mapped('date')
-            min_date = format_date(self.env, min(dates))
-            max_date = format_date(self.env, max(dates))
-            expense_name = min_date if max_date == min_date else "%s - %s" % (min_date, max_date)
+            if False not in dates:  # If at least one date isn't set, we don't set a default name
+                min_date = format_date(self.env, min(dates))
+                max_date = format_date(self.env, max(dates))
+                expense_name = min_date if max_date == min_date else "%s - %s" % (min_date, max_date)
```

Comment:
As we're moving the code, what about changing for a f-string? :nerd_face: 
```suggestion
                expense_name = min_date if max_date == min_date else f"{min_date} - {max_date}"
```